tests/integration/genutil/gentx_test.go

```diff
@@ -72,40 +72,47 @@ func (suite *GenTxTestSuite) SetupTest() {
 			configurator.AuthModule()),
 		simtestutil.DefaultStartUpConfig(),
 		&encCfg.InterfaceRegistry, &encCfg.Codec, &encCfg.TxConfig, &encCfg.Amino,
-		&suite.accountKeeper, &suite.bankKeeper, &suite.stakingKeeper)
-	suite.Require().NoError(err)
+		&f.accountKeeper, &f.bankKeeper, &f.stakingKeeper)
+	assert.NilError(t, err)
 
-	suite.ctx = app.BaseApp.NewContext(false, tmproto.Header{})
-	suite.encodingConfig = encCfg
-	suite.baseApp = app.BaseApp
+	f.ctx = app.BaseApp.NewContext(false, tmproto.Header{})
+	f.encodingConfig = encCfg
+	f.baseApp = app.BaseApp
 
 	amount := sdk.NewInt64Coin(sdk.DefaultBondDenom, 50)
 	one := math.OneInt()
-	suite.msg1, err = stakingtypes.NewMsgCreateValidator(
+	f.msg1, err = stakingtypes.NewMsgCreateValidator(
 		sdk.ValAddress(pk1.Address()), pk1, amount, desc, comm, one)
-	suite.NoError(err)
-	suite.msg2, err = stakingtypes.NewMsgCreateValidator(
+	assert.NilError(t, err)
+	f.msg2, err = stakingtypes.NewMsgCreateValidator(
 		sdk.ValAddress(pk2.Address()), pk1, amount, desc, comm, one)
-	suite.NoError(err)
+	assert.NilError(t, err)
+
+	return f
 }
 
-func (suite *GenTxTestSuite) setAccountBalance(addr sdk.AccAddress, amount int64) json.RawMessage {
-	acc := suite.accountKeeper.NewAccountWithAddress(suite.ctx, addr)
-	suite.accountKeeper.SetAccount(suite.ctx, acc)
+func setAccountBalance(t *testing.T, addr sdk.AccAddress, amount int64) json.RawMessage {
+	f := initFixture(t)
```

Comment:
couldn't you pass the fixture here instead in the function instead?
feel weird to see it initialized everytime this function is called.